numpy/lib/polynomial.py

```diff
@@ -1236,7 +1236,7 @@ def __init__(self, c_or_r, r=False, variable=None):
             raise ValueError("Polynomial must be 1d only.")
         c_or_r = trim_zeros(c_or_r, trim='f')
         if len(c_or_r) == 0:
-            c_or_r = NX.array([0.])
+            c_or_r = NX.array([0.], dtype=c_or_r.dtype)
```

Comment:
May as well make this an integer literal now too:
```suggestion
            c_or_r = NX.array([0], dtype=c_or_r.dtype)
```